fix: update starknet-burner package to match latest version (#3)

perpetual-cosmos opened the pull request:
Update starknet-burner Dependencies and Test Compatibility

Fixes #3 
close #3 

This pull request updates the starknet-burner package to align with the latest stable version of starknet.js (v6.24.1) and other dependencies as required for the scaffold-stark project. The changes include:





Updated starknet to ^6.24.1 in peerDependencies and devDependencies.



Updated @starknet-react/core to ^3.7.4 in peerDependencies and devDependencies.



Updated @starknet-react/chains to ^3.1.3 in peerDependencies and devDependencies.



Updated @starknet-io/types-js to ^0.8.4 in dependencies.



Tested compatibility using a standalone Node.js script with Starknet Devnet (shardlabs/starknet-devnet-rs) running via Docker. The script successfully:





Initialized BurnerConnector.



Connected to the devnet (chain ID: SN_SEPOLIA).



Retrieved the account address.



Queried the ETH balance of a pre-funded account.
![Screenshot 2025-06-06 195124](https://github.com/user-attachments/assets/f30781be-e622-4135-a677-1ebe3bdc25b7)

perpetual-cosmos committed:
fix: update starknet-burner package to match latest version (#3)

gianalarcon commented:
Hi, thanks for contributing. Suggest you to check this [repo/branch](https://github.com/Scaffold-Stark/scaffold-stark-2/blob/develop/packages/nextjs/package.json) to see what are the "latest" versions we using

perpetual-cosmos commented:
Thanks for the heads-up! I'll update it as soon as possible and make a commit shortly.

perpetual-cosmos committed:
fix: align dependencies with latest versions from scaffold-stark-2 develop branch

perpetual-cosmos commented:
Hi @gianalarcon , I've aligned the dependencies with the latest versions from the scaffold-stark-2 develop branch. Please have a look when you get a chance — looking forward to your feedback!

gianalarcon commented on package.json:
```diff
@@ -31,23 +31,28 @@
-    "starknet": "^6.12.0"
+    "@starknet-react/chains": "^3.1.3",
+    "@starknet-react/core": "^4.0.1-beta.1",
+    "@starknet-io/get-starknet-core": "^4.0.0",
```
Any reason why we adding this peerDependencies that we dont use it on this repo?

perpetual-cosmos committed:
removed @starknet-io/get-starknet-core from peerDependecies and devDependencies

perpetual-cosmos commented:
Hi @gianalarcon , 
thanks for the feedback! I reviewed the codebase and confirmed that `@starknet-io/get-starknet-core` is not used directly in `starknet-burner`, so I’ve removed it from `peerDependencies` and `devDependencies`. 

> I had added it while creating tests and experimenting with the library during my learning process, and I overlooked its inclusion—sorry about that!

 The other peer dependencies (`@starknet-react/chains`, `@starknet-react/core`, `starknet`, and `react`) are required for the burner wallet connector to integrate with Starknet and React applications, and their versions align with the `scaffold-stark-2` `nextjs` package ([repo/branch](https://github.com/Scaffold-Stark/scaffold-stark-2/blob/develop/packages/nextjs/package.json)). 

---
Thank you!!! Let me know if there’s anything else to address!

gianalarcon commented on package.json:
```diff
@@ -31,23 +31,26 @@
-    "starknet": "^6.12.0"
+    "@starknet-react/chains": "^3.1.3",
+    "@starknet-react/core": "^4.0.1-beta.1",
+    "react": "^18.0 || ^19.0.0",
```
why we need to add react?

gianalarcon commented on package.json:
```diff
@@ -31,23 +31,26 @@
-    "typescript": "^5.0.0"
+    "@starknet-react/chains": "^3.1.3",
+    "@starknet-react/core": "^4.0.1-beta.1",
+    "react": "^18.3.1",
```
why do we need to add react?

gianalarcon reviewed (changes requested):
Hi Sir, please test your solution, dont push whatever

gianalarcon commented:
Hi @perpetual-cosmos , do you still plan to contribute on this?

perpetual-cosmos commented:
Apologies for the delay. I’ve been experiencing some medical issues along with an increased workload at the office. I kindly request a bit more time to manage things.

I assure you that I’ll be able to get back to it positively by Saturday.

Thank you for your understanding.

gianalarcon committed:
Fix: Update depenencies

gianalarcon committed:
Fix: Remove unexisting test
